pkg/virt-handler/vm.go

```diff
@@ -2461,7 +2461,35 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationSource(origVMI *v1.Vir
 			d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is aborting migration.")
 		}
 	} else {
-		migrationConfiguration := d.clusterConfig.GetMigrationConfiguration()
+		migrationConfiguration := d.clusterConfig.GetMigrationConfiguration().DeepCopy()
+		// TODO: how to currently perform update here?
+		origVMI.Status.MigrationState.MigrationConfigSource = v1.ClusterWideConfig
+		vmi.Status.MigrationState.MigrationConfigSource = v1.ClusterWideConfig
+
+		vmiNamespace, err := d.clientset.CoreV1().Namespaces().Get(context.Background(), origVMI.Namespace, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		// Override cluster-wide migration configuration if migration policy exists
+		migrationList, err := d.clientset.MigrationPolicy().List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		matchedPolicy := migrationList.MatchPolicy(origVMI, vmiNamespace)
+
+		if matchedPolicy == nil {
+			log.Log.Object(vmi).Reason(err).Infof("no migration policy matched for VMI %s", vmi.Name)
+		} else {
+			isUpdated, err := matchedPolicy.GetMigrationConfByPolicy(migrationConfiguration)
+			if err != nil {
+				log.Log.Object(vmi).Reason(err).Warningf("cannot get migration config by migration policy")
+			} else if isUpdated {
+				origVMI.Status.MigrationState.MigrationConfigSource = v1.MigrationPolicyConfig
+				vmi.Status.MigrationState.MigrationConfigSource = v1.MigrationPolicyConfig
```

Comment:
what's going on here? The origVMI from the cache should never be modified, and why is both the origVMI and vmi getting modified here to begin with? THis isn't the update status function.

Also, why do we need this "MigrationConfigSource" status field?

Reply:
Fixed

Reply:
> Also, why do we need this "MigrationConfigSource" status field?

I've replaced that with `vmi.Status.MigrationState.MigrationPolicyName`. This seems valuable information to me, which can help to test and debug.